lib/elixir/lib/string.ex

```diff
@@ -1220,6 +1220,28 @@ defmodule String do
   end
 
   @doc ~S"""
+  Converts a numeric codepoint to a string containing the corresponding character.
+
+  ## Examples
+
+      iex> String.from_codepoint(97)
+      "a"
+
+      iex> String.from_codepoint(128_518)
+      "😆"
+
+      iex> String.from_codepoint("1F916")
+      "🤖"
+  """
+  def from_codepoint(cp) when is_integer(cp) when cp >= 0 and cp <= 2_097_151 do
```

Comment:
`2_097_151` is the maximum value representable with 21 bits, which is how many non-header bits are available in a four-byte UTF-8 string. In other words, this is the maximum UTF-8 value.

I don't especially like having it here as a "magic number", but I wasn't sure the best way to notate the meaning.